### PR TITLE
add add imu module

### PR DESCRIPTION
**Pull request by adambujak**

### CODE REVIEW

File: firmware/applications/SignBuddy/imu.c [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+  gpio_config.Pull = LL_GPIO_PULL_NO;
+  gpio_config.Alternate = LL_GPIO_AF_1;
+  LL_GPIO_Init(GPIOB, &gpio_config);
+
+  LL_I2C_InitTypeDef i2c_config = { 0 };
+  LL_I2C_EnableAutoEndMode(I2C1);
+  LL_I2C_DisableOwnAddress2(I2C1);
+  LL_I2C_DisableGeneralCall(I2C1);
+  LL_I2C_EnableClockStretching(I2C1);
+  i2c_config.PeripheralMode = LL_I2C_MODE_I2C;
```
Comment (adambujak) [inline firmware/applications/SignBuddy/imu.c:36]: increase speed

File: firmware/applications/SignBuddy/imu.c [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+#define PROCESS_PERIOD_MS    1000
+
+typedef struct {
+  uint32_t last_ticks;
+  i2c_t    i2c_instance;
+} state_t;
+
+static state_t s;
+
+static void hw_init(void)
```
Comment (stephen99scott) [inline firmware/applications/SignBuddy/imu.c:17]: Should this be in i2c.c?

**Review by stephen99scott — COMMENTED** [review]
Should the imu module be moved to the sensors module?

File: firmware/applications/SignBuddy/imu.c [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+{
+  LL_IOP_GRP1_EnableClock(LL_IOP_GRP1_PERIPH_GPIOB);
+  LL_APB1_GRP1_EnableClock(LL_APB1_GRP1_PERIPH_I2C1);
+
+  LL_GPIO_InitTypeDef gpio_config = { 0 };
+  gpio_config.Pin = LL_GPIO_PIN_6 | LL_GPIO_PIN_7;
```
Comment (stephen99scott) [inline firmware/applications/SignBuddy/imu.c:23]: Abstract pins and alternate function for consistency with other modules

File: firmware/applications/SignBuddy/imu.c (hunk 2 above)
Comment (adambujak) [reply firmware/applications/SignBuddy/imu.c:17]: no, I wanted it to be a generic config that you pass to i2c

File: firmware/applications/SignBuddy/imu.c (hunk 3 above)
Comment (adambujak) [reply firmware/applications/SignBuddy/imu.c:23]: yeah I forgot to make that an issue.. 

I'll just do it

Comment (stephen99scott) [reply firmware/applications/SignBuddy/imu.c:23]: Thanks

File: firmware/applications/SignBuddy/imu.c (hunk 2 above)
Comment (stephen99scott) [reply firmware/applications/SignBuddy/imu.c:17]: I was referring to the hw_init function, not sure if you were also referring to that

Comment (adambujak) [reply firmware/applications/SignBuddy/imu.c:17]: i was, the hw_init is just a generic function to initialize things related to hardware for the imu module. I want the i2c module to be module agnostic and not a single instance module, therefore the imu module should pass in a config that gets applied by the i2c module

Comment (stephen99scott) [reply firmware/applications/SignBuddy/imu.c:17]: I see now thanks for the clarification

**Review by stephen99scott — APPROVED** [review]

**Pull request closed without merging**